### ISSUE TITLE
Unuseful "hashAlg" capability for DSA / keyGen ?

### ISSUE THREAD

**Issue by davyrouillard** [cisco/libacvp#509]
Hi,
For tests on DSA / keyGen, we can see in the app_main.c the following parameter settings:
  ```
  …
    rv = acvp_cap_dsa_set_parm(ctx, ACVP_DSA_KEYGEN, ACVP_DSA_MODE_KEYGEN, ACVP_DSA_LN2048_224, ACVP_SHA224);
    CHECK_ENABLE_CAP_RV(rv);
    rv = acvp_cap_dsa_set_parm(ctx, ACVP_DSA_KEYGEN, ACVP_DSA_MODE_KEYGEN, ACVP_DSA_LN2048_224, ACVP_SHA256);
    CHECK_ENABLE_CAP_RV(rv);
    rv = acvp_cap_dsa_set_parm(ctx, ACVP_DSA_KEYGEN, ACVP_DSA_MODE_KEYGEN, ACVP_DSA_LN2048_224, ACVP_SHA384);
    CHECK_ENABLE_CAP_RV(rv);
    rv = acvp_cap_dsa_set_parm(ctx, ACVP_DSA_KEYGEN, ACVP_DSA_MODE_KEYGEN, ACVP_DSA_LN2048_224, ACVP_SHA512);
    CHECK_ENABLE_CAP_RV(rv);
…
```

whatever the value of the last parameter (ACVP_SHAxxx), the result remains the same :
`***ACVP [INFO][acvp_register:1971]--> [{"acvVersion":"1.0"},{"algorithms":[{"algorithm":"DSA","revision":"1.0","mode":"keyGen","prereqVals":[{"algorithm":"SHA","valValue":"same"},{"algorithm":"DRBG","valValue":"same"}],"capabilities":[{"l":2048,"n":224}]}]}]`

This is compliant with the [specification](https://usnistgov.github.io/ACVP/draft-fussell-acvp-dsa.html#section-7.5.1) that does not require a « hash function » capability for ACVP_DSA_KEYGEN (only relevant for DSA/pqgVer).
In that case, would it not be better to have another API to set the parameters or a « Non applicable » value?
Thanks.
Davy

**Comment by bfussell** [comment]
I agree it does not appear SHA is required. We probably just overloaded that API by using it for keygen.
We will take a look and remove any unnecessary registrations in app_main.

Thanks,

Barry

**Comment by bfussell** [comment]
Fixed by commits #510 and #511
